### PR TITLE
Week09

### PR DESCRIPTION
**Pull request by MdTanvirHossainTusher**

### CODE REVIEW

File: week09/200. Number of Islands.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// TC = O(N) where N = total no of cells in the grid
+// MC = O(1)
```
Comment (Sadman007) [inline week09/200. Number of Islands.cpp]: It is not `O(1)`. Recursion has it's memory usage.

File: week09/207. Course Schedule.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// TC = O(N + E)
+// MC = O(2N * N) where N is the total no of nodes
```
Comment (Sadman007) [inline week09/207. Course Schedule.cpp:2]: Shouldn't it be linear?

File: week09/419. Battleships in a Board.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                battleShips++;
+            }
+        }
+        return battleShips;
+    }
+};
+
+/**********************************/
+// TC = O(N) where N = total no of battle ships
+// MC = O(1) no extra memory used, but value changed
```
Comment (Sadman007) [inline week09/419. Battleships in a Board.cpp]: It is not `O(1)`. Recursion has it's memory usage.

**Review by Sadman007 — APPROVED** [review]
Need to be careful with complexity analysis. Overall, nice work! @MdTanvirHossainTusher

File: week09/207. Course Schedule.cpp (hunk 2 above)
Comment (MdTanvirHossainTusher) [reply week09/207. Course Schedule.cpp:2]: // TC = O(N) where N = total no of nodes
// MC = O(2*N) for two extra vector
// Auxiliary space  = O(N) ; for recursion call

Is it ok now?

**Pull request merged**